tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/OsgiBundleProject.java

```diff
@@ -204,6 +205,31 @@ private MavenProject getMavenProject(ReactorProject reactorProject) {
                 "Project not setup correctly");
     }
 
+    public Optional<ArtifactKey> readArtifactKey(ArtifactDescriptor descriptor) {
+        File location = descriptor.getLocation(true);
+        if (location == null) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(readArtifactKey(location));
+        } catch (OsgiManifestParserException e) {
+            logger.warn("First attempt failed, deleting file " + location + " :: " + e);
+            //something is wrong, delete the file and try to fetch it again!
+            if (!location.delete()) {
+                logger.warn("File can't be deleted!");
+            }
+            location = descriptor.getLocation(true);
+            if (location == null) {
+                return Optional.empty();
+            }
+            if (!location.exists()) {
```

Comment:
Rather https://docs.oracle.com/javase/7/docs/api/java/io/File.html#isFile() and the message should probably explain the most common cause:
- Path is a directory or
- File is not accessible to the current user